src/loan_form/LoanForm.jsx

```diff
@@ -29,14 +30,68 @@ export default class LoanForm extends React.Component {
     this.setState({ loans: newLoans })
   }
   
+  handleChange(id, name, value) {
+    const newLoans = this.state.loans
+
+    for (var i = 0; i < this.state.loans.length; i++) {
+      if (this.state.loans[i]['id'] == parseInt(id)) {
+        newLoans[i][camelCase(name)] = value
+        newLoans[i].monthsToPayOffLoan = this.findLoanLength(newLoans[i])
+      }
+    }
```

Comment:
ah now that this is smaller, i can better understand it and give _yet another comment_.

`Array.prototype.map` and `Array.prototype.forEach` in javascript each take a function that accepts _at least_ one argument, the value at each stage of iteration. However, these functions _can_ [accept more arguments](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#Parameters). Notably the *index* of the item will be passed as the second argument to any 2-arity function. _So_ in this case you could `map` over loans and use the index to update the new loans array.

_However_ it's worth noting, that by setting `newLoans` to `this.state.loans` you are _not_ copying the value in the state, you are just defining a new name to reference that value. If you were to map over `this.state.loans` you would actually get a new array (which is what i assume you want to do since you're doing the assignment in the first place).

You might want something like this here:
```javascript
const newLoans = this.state.loans.map(loan => {
  if (loan.id === parseInt(id)) { // remember to use === here so we don't get weird casting behavior
    const newLoan = {
      ...loan, // this will dump any key/value pairs from loan into this new object
      [camelCase(name)]: value, // this will create the appropriate key and insert it
    }
    newLoan.monthsToPayOffLoan = this.findLoanLength(newLoan)

    return newLoan
  } 
  else return loan
})

Reply:
although it kind of feels like `findLoanLength` is little out of place here? like maybe we should use in `calculate` and then just call calculate at the end of this method.